tap_xero/client.py

```diff
@@ -87,15 +152,41 @@ def refresh_credentials(self, config, config_path):
             "refresh_token": config["refresh_token"],
         }
         resp = self.session.post("https://identity.xero.com/connect/token", headers=headers, data=post_body)
-        resp.raise_for_status()
-        resp = resp.json()
 
-        # Write to config file
-        config['refresh_token'] = resp["refresh_token"]
-        update_config_file(config, config_path)
-        self.access_token = resp["access_token"]
-        self.tenant_id = config['tenant_id']
+        if resp.status_code != 200:
+            raise_for_error(resp)
+        else:
+            resp = resp.json()
+
+            # Write to config file
+            config['refresh_token'] = resp["refresh_token"]
+            update_config_file(config, config_path)
+            self.access_token = resp["access_token"]
+            self.tenant_id = config['tenant_id']
+
+
+    def check_platform_access(self, config, config_path, check_authentication=True):
```

Comment:
Code added w.r.to TDL-594 (Check authentication and authorization during discovery mode)